lib/src/ast_extensions.dart

```diff
@@ -390,6 +390,8 @@ extension PatternExtensions on DartPattern {
   bool get canBlockSplit => switch (this) {
         ListPattern(:var elements, :var rightBracket) =>
           elements.canSplit(rightBracket),
+        MapPattern(:var elements, :var rightBracket) =>
```

Comment:
Do both variable assignments have the same type? Consider using `||` to merge the two cases.

Reply:
They don't, sadly. `ListPatternElement`s and `MapPatternElement`s. Thanks for the reminder though.